### PR TITLE
wire printflags through run cmd

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Adds PrintFlags to `run` command.

cc @soltysh @deads2k

### CODE REVIEW

**Comment by deads2k** [comment]
/assign @soltysh 
@kubernetes/sig-cli-maintainers

File: hack/make-rules/test-cmd-util.sh [hunk 1]
```diff
@@ -1296,9 +1296,9 @@ run_kubectl_run_tests() {
   kubectl run nginx-extensions "--image=$IMAGE_NGINX" "${kube_flags[@]}"
   # Post-Condition: Deployment "nginx" is created
   kube::test::get_object_assert deployment.extensions "{{range.items}}{{$id_field}}:{{end}}" 'nginx-extensions:'
-  # and old generator was used, iow. old defaults are applied
+  # new generator was used
   output_message=$(kubectl get deployment.extensions/nginx-extensions -o jsonpath='{.spec.revisionHistoryLimit}')
-  kube::test::if_has_not_string "${output_message}" '2'
+  kube::test::if_has_string "${output_message}" '2'
```
Comment (juanvallejo) [inline hack/make-rules/test-cmd-util.sh]: @soltysh this was updated, as this PR defaults to using the new generator

File: pkg/kubectl/cmd/run.go [hunk 2]
```diff
@@ -141,9 +179,43 @@ func addRunFlags(cmd *cobra.Command) {
...
+		return printer.PrintObj(obj, o.Out)
+	}
+
+	return nil
+}
+
+func (o *RunOpts) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
 	// Let kubectl run follow rules for `--`, see #13004 issue
-	if len(args) == 0 || argsLenAtDash == 0 {
+	if len(args) == 0 || o.ArgsLenAtDash == 0 {
```
Comment (soltysh) [inline pkg/kubectl/cmd/run.go]: Once this merges, we should clean this and move them to `Validate`, but in a separate PR.

**Review by soltysh — APPROVED** [review]
/approve no-issue

**Comment by soltysh** [comment]
Rebase and I'll apply the lgtm label

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**